From 06c43fd81baf70bed6a3729c0e644769da8b14e1 Mon Sep 17 00:00:00 2001
From: Mark Harmstone <hellas@burntcomma.com>
Date: Sun, 14 Dec 2014 23:45:41 +0000
Subject: msdmo: Fix invalid usage of RegOpenKeyExW.

Patch to fix typo in msdmo. RegOpenKeyExW returns 0 on success, and non-zero on failure.
---
 dlls/msdmo/dmoreg.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/dlls/msdmo/dmoreg.c b/dlls/msdmo/dmoreg.c
index e8f67c5..ff0e8f0 100644
--- a/dlls/msdmo/dmoreg.c
+++ b/dlls/msdmo/dmoreg.c
@@ -406,7 +406,7 @@ static HRESULT IEnumDMO_Constructor(
     /* If not filtering by category enum from media objects root */
     if (IsEqualGUID(guidCategory, &GUID_NULL))
     {
-        if (!RegOpenKeyExW(HKEY_CLASSES_ROOT, szDMORootKey, 0, KEY_READ, &lpedmo->hkey))
+        if (RegOpenKeyExW(HKEY_CLASSES_ROOT, szDMORootKey, 0, KEY_READ, &lpedmo->hkey))
             hr = E_FAIL;
     }
     else
@@ -416,7 +416,7 @@ static HRESULT IEnumDMO_Constructor(
 
         wsprintfW(szKey, szCat3Fmt, szDMORootKey, szDMOCategories,
             GUIDToString(szguid, guidCategory));
-        if (!RegOpenKeyExW(HKEY_CLASSES_ROOT, szKey, 0, KEY_READ, &lpedmo->hkey))
+        if (RegOpenKeyExW(HKEY_CLASSES_ROOT, szKey, 0, KEY_READ, &lpedmo->hkey))
             hr = E_FAIL;
     }
 
-- 
2.2.1

